sklearn/utils/estimator_checks.py

```diff
@@ -479,9 +479,6 @@ def _set_checking_parameters(estimator):
         # K-Means
         estimator.set_params(n_init=2)
 
-    if hasattr(estimator, "n_components"):
```

Comment:
I could assume that it was to speed up the tests. Is there a reason to remove it?

Reply:
We used to raise a warning, but now we error so tests would fail because in some cases 2 was too high.

Since setting it to 1 is not really meaningful, I left it to the default.

Reply:
In terms of testing time, `pytest sklearn/tests/test_common.py -k LinearDiscriminantAnalysis` goes from 4.9 secs to 5.2 secs on my machine so that's negligible